github/github.go

```diff
@@ -64,11 +65,8 @@ type Client struct {
 	// User agent used when communicating with the GitHub API.
 	UserAgent string
 
-	// Rate specifies the current rate limit for the client as determined by the
-	// most recent API call.  If the client is used in a multi-user application,
-	// this rate may not always be up-to-date.  Call RateLimits() to check the
-	// current rate.
-	Rate Rate
+	rateMu sync.Mutex
```

Comment:
no, this is fine.

Though I would certainly be curious to see benchmarks of the two, not just for this library but in general.  I just wonder how much difference it really makes in practice.